dev/com.ibm.ws.managedobject/src/com/ibm/ws/managedobject/internal/ManagedObjectFactoryImpl.java

```diff
@@ -43,15 +40,24 @@ public boolean managesInjectionAndInterceptors() {
 
     @Override
     public Class<T> getManagedObjectClass() {
-        return constructor.getDeclaringClass();
+        return managedClass;
     }
 
     /**
      * Returns the constructor that will be used by this factory to create the managed object.
+     *
+     * @throws ManagedObjectException
      */
     @Override
-    public Constructor<T> getConstructor() {
-        return constructor;
+    public Constructor<T> getConstructor() throws ManagedObjectException {
+        if (this.constructor == null) {
+            try {
+                this.constructor = this.managedClass.getConstructor((Class<?>[]) null);
+            } catch (NoSuchMethodException e) {
+                throw new ManagedObjectException(e);
```

Comment:
should this have an NLS message?

Reply:
And why was this change needed?  Is the issue that some object types don't require a default constructor? But this particular class is only used when a default constructor is required.

Reply:
I spoke to the NLS team and they didn't find a rule saying it *has* to, but I think this one should do, since it's something the customer can potentially fix.